packages/build/src/plugins_core/build_command.js

```diff
@@ -38,6 +38,9 @@ const coreCommand = async function ({
     handleBuildCommandOutput(buildCommandOutput, logs)
     return {}
   } catch (error) {
+    // The child process `error.message` includes stderr and stdout output which most of the times contains duplicate
+    // information. We rely on `error.shortMessage` instead.
```

Comment:
I am wondering whether we should mention instead that the build command stdout/stderr is streamed, which is why including it in the error message as well would lead to duplicate logs?

Reply:
Is it a matter of streaming? 🤔 I just assumed this was related with the `message` field contents, which we always log(?). Anyway happy to rephrase the comment to something that makes more sense 👍

Reply:
In production (i.e. when `buffer: false` CLI flag is used), we use `stdio: 'inherit'` on the build command. This prevents Node.js `child_process` from being able to buffer stdout/stderr and as a consequence, `error.message` should not contain stdout/stderr and be equal to `error.shortMessage`.

In tests, we pass `buffer: true` so that logs are concatenated to a `logs` array instead of being printed on stdout/stderr. We also then pass `stdio: 'pipe'` on the build command, which allows buffering. At the end of each test, we print the contents of those logs. I.e. without `error.shortMessage`, we would end up printing the build command output twice.

Reply:
Ah makes perfect sense, thanks for taking the time to explain it. I've updated the comment with what you've just shared in https://github.com/netlify/build/pull/3440/commits/75fbf286b41f2c637d1f1f48097b2ba20dcd918d 🙌